skfda/misc/scoring.py

```diff
@@ -125,6 +125,37 @@ def _multioutput_score_grid(
     return float(np.mean(score.integrate()[0]) / _domain_measure(score))
 
 
+def _integral_average_fdatairregular(
+    score: FDataIrregular,
+    squared: bool = True,
+    weights: NDArrayFloat | None = None,
+) -> float:
+    """Calculate the weighted average of the normalized integrals of the score.
+
+    The integral of the score is normalized because each integral is divided by
+    the length of the curve's domain.
+
+    If the score is vector-valued, then the mean of each codimension integral
+    is calculated for every functional observation.
+
+    Args:
+        score: Score of the functions.
+        squared: If False, the square root is taken.
+        weights: Weights for the mean.
+    """
+    if score.dim_domain != 1:
+        raise ValueError(
+            "Only univariate FDataIrregular objects are supported",
+        )
+    if not squared:
+        score = np.sqrt(score)
```

Comment:
I do not know the reasoning behind this design, but this is what is done for `FDataGrid`s: https://github.com/GAA-UAM/scikit-fda/blob/a5b7617f6cad9d70e2104fe8fee352e1adf4e662/skfda/misc/scoring.py#L110-L125

Reply:
We need to understand this and compare it with the multivariate behavior in scikit-learn.

Reply:
It seems like ``squared`` is a deprecated parameter in [``sklearn.metrics.mean_squared_error``](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html), it was meant to be used for computing the Root Mean Squared Error (RMSE), when ``squared=False``, instead of the Mean Squared Error (MSE), which is the default: ``squared=True``. The parameter name is confusing because it seems to refer to the _Squared_ part of the score name instead of the _Root_.

The RMSE for multivariate variables is defined as 

$$RMSE = \sqrt{\frac 1 n \sum_{i=1}^n (\hat y_i - y_i)^2}$$
![image](https://github.com/GAA-UAM/scikit-fda/assets/71875712/8c642de8-ca5a-4a2e-b4e2-d401411cc442)

In the `FData` cases, we integrate that value for each point in the domain. So taking square root before integrating is correct:

$$RMSE = \int_D \ \ \sqrt{\frac 1 n \sum_{i=1}^n \left(\hat y_i(t) - y_i(t)\right)^2}\ \ dt$$

<!--- ![image](https://github.com/GAA-UAM/scikit-fda/assets/71875712/c85f08dd-8732-459e-82fc-163adcf5c4f4) --->

Reply:
I see.